ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/identity_management_service.proto

```diff
@@ -0,0 +1,138 @@
+  // daml-on-sql: returns an identifier supplied on command line at launch time
+  // daml-on-sawtooth: as above
+  // canton: returns globally unique identifier
+  rpc GetParticipantId (GetParticipantIdRequest) returns
+    (GetParticipantIdResponse);
+
+  // List the participants known by the backing participant.
+  // TODO: Check what the use case is for listing all participants
+  // Admin tooling in daml-on-sawtooth knows the topology from other sources
+  // daml-on-sql knows only one participant. Suggest to remove it
+  rpc ListKnownParticipants (ListKnownParticipantsRequest) returns
+    (ListKnownParticipantsResponse);
+
+  // List the parties known by the backing participant.
+  // The list returned contains parties whose ledger access is facilitated by
+  // backing participant and teh ones maintained elsewhere
+  // TODO: Check what the use case is for listing non-local parties
+  rpc ListKnownParties (ListKnownPartiesRequest) returns
+    (ListKnownPartiesResponse);
+
+  // Adds a new party to the set managed by the backing participant.
+  // Caller specifies a party identifier suggestion, the actual identifier
+  // allocated might be different and is implementation specific.
+  // daml-on-sql: suggestion's uniqueness is checked and call rejected if the
+  // identifier is already present
+  // daml-on-sawtooth: suggestion is prefixed with backing participant
+  // identifier and the composite is tested for uniqueness
+  // canton: completely different globally unique identifier is allocated.
+  // Behind the scenes calls to an internal protocol are made. As that protocol
+  // is richer than the the surface protocol, the arguments take implicit values
+  // Future extension: augment the request with additional meta-data containing
+  // human readable party name, privileges, organization etc.
+  rpc AllocateParty (AllocatePartyRequest) returns (AllocatePartyResponse);
+}
+
+message GetParticipantIdRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message GetParticipantIdResponse {
+  ParticipantDetails participant_details = 1;
+}
+
+message ParticipantDetails {
+
+  string participant_id = 1;
+}
+
+message ListKnownParticipantsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownParticipantsResponse {
+  repeated ParticipantDetails participant_details = 1;
+}
+
+message ListKnownPartiesRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification
+  // Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListKnownPartiesResponse {
+
+  // The details of all DAML parties hosted by the participant.
+  // Required
+  repeated PartyDetails party_details = 1;
+}
+
+message PartyDetails {
+
+  // The ID of a DAML party hosted by the participant.
+  // Required
+  string party_id = 1;
```

Comment:
As said, please don't use party_id but extend this with more meaningful information. I think assuming that we can get away with a string is just not a good idea.

Reply:
Can't answer to your above comment wrt party id. So answering here, which is related.

What do you think about the following approach:
```
message PartyDetails {
  // Stable identifier for this party. Will never change and is not necessarily human readable.
  string party_id 
  // Human readable name for that party
  string display_name
}
```
I'm a bit wary of adding free-form attribute sets, as their format tends to get fixed over time without much control over the end result. Can be convinced of adding something like the following though:  
```
map<string, string> party_attributes
```
@daravep what do you think?

Reply:
@daravep, I don't understand what you mean when you say "please don't use party_id". I surely must be able to get hold of the one and only stable identifier that I can use over the ledger api and for which I can issue access tokens. Do you mean, don't use party_id alone?

Reply:
Yes, I mean don't use `party_id` as a term. I think that it simplifies the underlying problem too much. At the end, what we have is a "stable identifier" and a "volatile human readable identifier".

Reply:
How about Simon's idea, that addresses you concern doesn't it? I have checked elsewhere in our api's and it seems that we use the term "party" to describe the same semantic entity elsewhere:
https://github.com/digital-asset/daml/blob/af1bf0cd6735dd0d1faf2fe76e91f344409c2f23/ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/commands.proto#L40
and here:
https://github.com/digital-asset/daml/blob/f9839a7e4cc6daaf36cafbf469610bb555292f33/ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto#L50